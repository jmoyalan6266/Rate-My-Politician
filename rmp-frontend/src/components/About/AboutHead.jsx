import { Jumbotron } from 'react-bootstrap';

const AboutHead = () => {
    return (
        <div>
            <h1 className="my-4">About</h1>
            <Jumbotron>
                <h3>
                    Rate My Politician allows you to see how politicans are
                    being covered in the media.
                </h3>
                <p>
                    Our site gathers information about news sources, states, and
                    politicians across the United States so that voters and
                    residents can learn how their state or representatives are
                    being covered in different forms of media, including in news
                    articles and on twitter.
                </p>
            </Jumbotron>
        </div>
    );
};

export default AboutHead;
