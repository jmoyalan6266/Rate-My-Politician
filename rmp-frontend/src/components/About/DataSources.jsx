import { Card, Container, Row } from "react-bootstrap";
import news_api from "./images/tools/news_api.png";
import google_civic from "./images/tools/google_civic.png";
import twitter from "./images/tools/twitter.jpg";
import data_usa from "./images/tools/data_usa.jpg";
import bigpicture from "./images/tools/bigpicture.jpg";
import civil_services from "./images/tools/civil_services.jpg";

const DataSources = () => {
    return (
        <div>
            <h2>Data Sources</h2>
            <Container className="mt-4">
                <Row sm>
                    <Card
                        as="a"
                        href="https://newsapi.org"
                        style={{
                            width: "12rem",
                        }}
                        className="mx-auto text-decoration-none text-dark"
                    >
                        <Card.Img src={news_api} variant="top" />
                        <Card.Body>
                            <Card.Title>News API</Card.Title>
                            <Card.Text>
                                Used to scrape news sources and show articles.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card
                        as="a"
                        href="https://developers.google.com/civic-information"
                        style={{
                            width: "12rem",
                        }}
                        className="mx-auto text-decoration-none text-dark"
                    >
                        <Card.Img src={google_civic} variant="top" />
                        <Card.Body>
                            <Card.Title>Google Civic API</Card.Title>
                            <Card.Text>
                                Used to scrape data about politicians across
                                states.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card
                        as="a"
                        href="https://bigpicture.io/docs/enrichment/company/"
                        style={{
                            width: "12rem",
                        }}
                        className="mx-auto text-decoration-none text-dark"
                    >
                        <Card.Img src={bigpicture} variant="top" />
                        <Card.Body>
                            <Card.Title>BigPicture Company API</Card.Title>
                            <Card.Text>
                                Used to scrape company data for news sources.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card
                        as="a"
                        href="https://datausa.io/about/api/"
                        style={{
                            width: "12rem",
                        }}
                        className="mx-auto text-decoration-none text-dark"
                    >
                        <Card.Img src={data_usa} variant="top" />
                        <Card.Body>
                            <Card.Title>Data USA API</Card.Title>
                            <Card.Text>
                                Used to scrape states and attribute data.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card
                        as="a"
                        href="https://github.com/CivilServiceUSA/us-states"
                        style={{
                            width: "12rem",
                        }}
                        className="mx-auto text-decoration-none text-dark"
                    >
                        <Card.Img src={civil_services} variant="top" />
                        <Card.Body>
                            <Card.Title>Civil Services</Card.Title>
                            <Card.Text>
                                Used to obtain state flags, maps, and state
                                images.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                </Row>
            </Container>
        </div>
    );
};

export default DataSources;
