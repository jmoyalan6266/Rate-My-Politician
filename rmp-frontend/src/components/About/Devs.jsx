import { Card, Badge, Container, Row, Col } from "react-bootstrap";
import laura from "./images/devs/laura.jpeg";
import kent from "./images/devs/kent.png";
import joseph from "./images/devs/joseph.jpg";
import preeti from "./images/devs/preeti.jpg";
import rachel from "./images/devs/rachel.jpg";

const Devs = ({ devs }) => {
    const images = [laura, kent, joseph, preeti, rachel];
    const bios = [
        "Laura is in her last semester at UT. Senioritis is really taking over her. After graduation, she’ll be moving to the Dallas area for her job and to be close to her family. Until then, she’ll be working on the front end of this project.",
        "I'm a junior Computer Science and Geography major also pursuing a Smart Cities BDP Certificate. I'm from Austin and enjoy learning about Web Development, hiking, and playing music.",
        "Hi I’m Joseph! I'm a junior majoring in Computer Science and Math and want to pursue a job in the world of FinTech. I'm from Plano, Texas and I enjoy gaming, playing volleyball, listening to music, and spending time with my friends in my free time.",
        "Hi! I'm a junior studying Computer Science and pursuing a Business Minor, as well as a Certificate in Applied Statistical Modeling. I'm from Pittsburgh, Pennsylvania, and in my free time, I enjoy photography and watching movies!",
        "I'm a junior who is new to web development. I'm from Houston and I'm an avid reader who is trying to safely knock things off of my Austin bucket list.",
    ];
    const tests = [11, 11, 11, 11, 10];

    return (
        <div>
            <h2 className="mb-3">Developers</h2>
            <Container>
                <Row>
                    {devs.map((dev, i) => (
                        <Col key={i} sm='auto' className='mb-4 mx-auto'>
                            <Card style={{width: '20rem'}}  className='h-100'>
                                <Card.Img
                                    variant="top"
                                    src={images[i]}
                                ></Card.Img>
                                <Card.Title className="p-3 mx-auto">
                                    {dev.name}
                                </Card.Title>
                                <Card.Text className="mx-4 mb-4">
                                    {bios[i]}
                                </Card.Text>
                                <Card.Text className="px-3 pb-2 mx-auto">
                                    Commits:{" "}
                                    <Badge
                                        className="p-2 mb-2"
                                        variant="primary"
                                    >
                                        {dev.commits}
                                    </Badge>
                                    <br />
                                    Issues:{" "}
                                    <Badge
                                        className="p-2 mb-2"
                                        variant="primary"
                                    >
                                        {dev.issues}
                                    </Badge>
                                    <br />
                                    Tests:{" "}
                                    <Badge
                                        className="p-2 mb-2"
                                        variant="primary"
                                    >
                                        {tests[i]}
                                    </Badge>
                                </Card.Text>
                            </Card>
                        </Col>
                    ))}
                </Row>
            </Container>
        </div>
    );
};

export default Devs;
