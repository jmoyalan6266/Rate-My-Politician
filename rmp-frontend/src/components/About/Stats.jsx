import { Table } from "react-bootstrap"

const Stats = ({ numCommits, numIssues }) => {
    return (
        <div>
            <h2>Project Stats</h2>
            <Table className="m-4" bordered striped responsive="xl">
                <tbody>
                    <tr>
                        <td>Commits</td>
                        <td>Issues</td>
                        <td>Tests</td>
                    </tr>
                    <tr>
                        <td>{numCommits}</td>
                        <td>{numIssues}</td>
                        <td>54</td>
                    </tr>
                </tbody>
            </Table>
        </div>
    ); 
};

export default Stats;