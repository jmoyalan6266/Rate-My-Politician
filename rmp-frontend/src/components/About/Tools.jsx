import { Container, Row, Card, Button } from "react-bootstrap";
import gitlab from "./images/tools/gitlab.jpg";
import postman from "./images/tools/postman.jpg";
import react from "./images/tools/react.png";
import react_bootstrap from "./images/tools/react_bootstrap.png";
import jupyter from "./images/tools/jupyter.png";
import nivo from "./images/tools/nivo.png";
import aws from "./images/tools/aws.jpg";
import flask from "./images/tools/flask.png";

const Tools = () => {
    return (
        <div>
            <h2>Tools</h2>
            <Container className="my-4">
                <Row sm>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={gitlab} variant="top" />
                        <Card.Body>
                            <Card.Title>Gitlab</Card.Title>
                            <Card.Text>
                                Repository hosting used for version control.
                            </Card.Text>
                            <Button
                                href="https://gitlab.com/cs-373-group-7/rate-my-politician"
                                size="sm"
                            >
                                Our Repo
                            </Button>
                        </Card.Body>
                    </Card>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={postman} variant="top" />
                        <Card.Body>
                            <Card.Title>Postman</Card.Title>
                            <Card.Text>
                                REST API design and testing tool.
                            </Card.Text>
                            <Button
                                href="https://documenter.getpostman.com/view/12075941/Tz5jeLVC"
                                size="sm"
                            >
                                Our API docs
                            </Button>
                        </Card.Body>
                    </Card>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={react} variant="top" />
                        <Card.Body>
                            <Card.Title>React</Card.Title>
                            <Card.Text>
                                Javascript frontend framework.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={react_bootstrap} variant="top" />
                        <Card.Body>
                            <Card.Title>React Bootstrap</Card.Title>
                            <Card.Text>
                                Javascript library for styled components in
                                react.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={jupyter} variant="top" />
                        <Card.Body>
                            <Card.Title>Jupyter Lab</Card.Title>
                            <Card.Text>
                                Jupyter Lab is a web interface for writing
                                Jupyter notebooks, used for API data scraping.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                </Row>
                <Row className="my-4">
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={nivo} variant="top" />
                        <Card.Body>
                            <Card.Title>Nivo</Card.Title>
                            <Card.Text>
                                Graphing library built with D3.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={aws} variant="top" />
                        <Card.Body>
                            <Card.Title>AWS</Card.Title>
                            <Card.Text>
                                Cloud platform for backend and frontend
                                deployment.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                    <Card style={{ width: "13rem" }} className="mx-auto">
                        <Card.Img src={flask} variant="top" />
                        <Card.Body>
                            <Card.Title>Flask</Card.Title>
                            <Card.Text>
                                Python backend framework for REST API
                                implementation.
                            </Card.Text>
                        </Card.Body>
                    </Card>
                </Row>
            </Container>
        </div>
    );
};

export default Tools;
