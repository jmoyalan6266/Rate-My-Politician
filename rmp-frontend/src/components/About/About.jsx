import { Container, Spinner } from "react-bootstrap";
import useStatistics from "../../hooks/useStatistics";
import Devs from "./Devs";
import AboutHead from "./AboutHead";
import Stats from "./Stats";
import DataSources from "./DataSources";
import Tools from "./Tools";
import Loading from "../Loading";

const About = () => {
    const { devs, numCommits, numIssues } = useStatistics();
    const allDevs = devs?.filter((dev) => dev.commits > 0);
    return (
        <Container>
            <AboutHead />
            {!devs ? (
                <Loading />
            ) : (
                <Devs devs={allDevs} />
            )}
            <hr className="mt-5" />
            <Stats numCommits={numCommits} numIssues={numIssues} />
            <hr className="mt-5" />
            <DataSources />
            <hr className="mt-5" />
            <Tools />    
            <hr className="mt-5" />
            <a href="https://www.youtube.com/watch?v=AtbX7xlNhQM"><h2 className="mb-5">Presentation Video</h2></a>
        </Container>
    );
};

export default About;
