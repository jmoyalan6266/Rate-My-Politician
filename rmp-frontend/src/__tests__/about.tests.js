import React from "react"
import { configure, shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import About from "../components/About/About"

configure({ adapter: new Adapter() })

describe("Test About page views", () => {
    test("About", () => {
        const aboutTest = shallow(<About />)
        expect(aboutTest).toMatchSnapshot()
    })
})
