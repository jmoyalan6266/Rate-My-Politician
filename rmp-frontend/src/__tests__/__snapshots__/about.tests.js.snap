// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test About page views About 1`] = `
<Container
  fluid={false}
>
  <AboutHead />
  <Loading />
  <hr
    className="mt-5"
  />
  <Stats />
  <hr
    className="mt-5"
  />
  <DataSources />
  <hr
    className="mt-5"
  />
  <Tools />
</Container>
`;
